edit-post/components/plugin-sidebar/index.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, Children, cloneElement, compose } from '@wordpress/element';
+import { Slot, Fill, withFocusReturn } from '@wordpress/components';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { withPluginContext } from '@wordpress/plugins';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import SidebarLayout from './sidebar-layout';
+import ErrorBoundary from './sidebar-error-boundary';
+
+/**
+ * Name of slot in which the sidebar should fill.
+ *
+ * @type {String}
+ */
+const SLOT_NAME = 'PluginSidebar';
+
+/**
+ * The plugin sidebar fill.
+ */
+class PluginSidebar extends Component {
+	constructor( props ) {
+		super( props );
+
+		if ( typeof props.name !== 'string' ) {
+			throw 'Sidebar names must be strings.';
+		}
+		if ( ! /^[a-z][a-z0-9-]*$/.test( props.name ) ) {
+			throw 'Sidebar names must include only lowercase alphanumeric characters or dashes, and start with a letter. Example: "my-sidebar".';
+		}
+	}
+
+	/**
+	 * Generates the UI plugin identifier by combining the plugin name and the sidebar name.
+	 *
+	 * Also registers the plugin in the plugin registry.
+	 */
+	componentWillMount() {
+		if ( ! this.namespacedName ) {
+			this.namespacedName = `${ this.props.pluginContext.namespace }/${ this.props.name }`;
+		}
+		this.props.pluginContext.registerUIComponent( this.namespacedName, 'sidebar' );
```

Comment:
I'm a bit confused by this. This seems to be recreating the same behavior for which Slot/Fill exists to provide. Namely, if we want to register a component for use elsewhere, it simply renders itself into a given fill. Why is this not occupying the general "Sidebar" fill.

Seems like PluginSidebar would both fill the generic sidebar, and also provide its own slot for plugins to fill it.

Reply:
The problem is that I need a way to check whether an id for a sidebar corresponds to an id for a registered plugin sidebar.

Reply:
In the `isPluginSidebarOpened` selector to be specific.